### PR TITLE
Resolve incident explicit with command request

### LINKED ISSUE #1033: Reiterate on incident concept
Incidents are currently not easy to use. The major pain point is that it is not easy to say what has to be done in order to resolve the incident. Things we could chanage:

* Do incidents need an event lifecycle?
* Do incidents need a separate identity (key)?

Ideas:

* Add a new record type `INCIDENT`
* When event processing fails, we create a new event that is an exact copy but has the type `INCIDENT`.
* Whenever we receive a command that updates this entity, we try to reprocess the incident event
* This needs some more thought, e.g. go through incident creation and resolution step by step assuming the users perspective

### LINKED ISSUE #1649: I can use the java client API to resolve an incident
An user should be able to resolve an incident by key.

### LINKED ISSUE #1651: There exist on broker and gateway side an API to resolve an incident by key

### PR DESCRIPTION
**Pull request by ChrisKujawa**
The incident resolving for message subscriptions had the problem that in CatchEventOutput the side effect is overwritten. I hand crafted shortly something, which can replaced when #1680 is merged (SideEffectQueue). But it works for now and can be reviewed.

Added resolve incident command to the gateway protocol and to the java client.
The broker and qa tests are also using the resolve method. There is no inplicit incident resolving anymore.

closes #1651 
closes #1649 
closes #1033

### CODE REVIEW

**Comment by Zelldon** [comment]
@saig0 rebased to develop and used the side effect queue instead of the hack :sweat_smile: It seems there is a deadlock in the ElasticsearchExporter I dont know why :man_shrugging:

**Review by saig0 — CHANGES_REQUESTED** [review]
:+1:

File: exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchExporterIT.java [hunk 1]
```diff
@@ -117,8 +119,16 @@ public void shouldExportRecords() {
               if (fail.getAndSet(false)) {
                 // fail job
                 client.newFailCommand(job.getKey()).retries(0).errorMessage("failed").send().join();
+
+                final Record<IncidentRecordValue> incident =
+                    RecordingExporter.incidentRecords(IncidentIntent.CREATED).getFirst();
                 // update retries to resolve incident
                 client.newUpdateRetriesCommand(job.getKey()).retries(3).send().join();
+                zeebeClient
```
[the remark this answers is not in the document]
Comment (Zelldon) [reply exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchExporterIT.java]: I aggree on this, maybe i find a better solution

File: broker-core/src/test/java/io/zeebe/broker/incident/ExpressionIncidentTest.java [hunk 2]
```diff
@@ -197,13 +199,15 @@ public void shouldResolveIncidentForFailedConditionAfterUploadingWrongPayload()
     testClient.createWorkflowInstance("workflow", asMsgPack("foo", "bar"));
 
     // then incident is created
-    testClient.receiveFirstIncidentEvent(IncidentIntent.CREATED);
+    final Record<IncidentRecordValue> incidentEvent =
```
[the remark this answers is not in the document]
Comment (Zelldon) [reply broker-core/src/test/java/io/zeebe/broker/incident/ExpressionIncidentTest.java]: Yeah maybe we can split it

**Review by saig0 — APPROVED** [review]

**Pull request merged**